docker/build-tools/Dockerfile

```diff
@@ -93,7 +93,8 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     gnupg2 \
     software-properties-common \
     unzip \
-    xz-utils
+    xz-utils \
+    && apt-get upgrade -y
```

Comment:
I cannot find a conclusive source but i have seen a lot of discussion that this is an anti-pattern. Shouldn't the `ubuntu:focal` image base be updated?

Reply:
`upgrade` was an anti-pattern at one point, but that has been relaxed. It was removed in the Docker docs here: https://github.com/docker/docker.github.io/pull/12571.

Reply:
As for getting past the linter, we might want to move it up. Hadolint 2.2.0 should fix this, of course there is a chicken-egg problem. Not sure what else the linter might find after moving up.

Reply:
Also, I would expect that this would be done in the final images like build_tools, etc.

Reply:
Also, I see now: `Official Debian and Ubuntu images automatically run apt-get clean, so explicit invocation is not required.` so that could be removed.